### PR TITLE
difference_delta function

### PR DESCRIPTION
**Pull request by leosartaj**
This PR introduces `DifferenceDelta` function, somewhat similar to what [wolfram](https://reference.wolfram.com/language/ref/DifferenceDelta.html) has.

@jcrist @flacjacket

### CODE REVIEW

**Comment by leosartaj** [comment]
There might be need to implement some more ``_eval_differenceDelta`` functions, which can be done at a later stage.

File: sympy/core/expr.py [hunk 1]
```diff
@@ -2845,6 +2845,46 @@ def diff(self, *symbols, **assumptions):
         assumptions.setdefault("evaluate", True)
         return Derivative(self, *new_symbols, **assumptions)
 
+    def differenceDelta(self, n=None, step=1):
```
Comment (jcrist) [inline sympy/core/expr.py]: Common python convention is to use snake_case instead of camelCase. So this should be `difference_delta`.

Comment (leosartaj) [inline sympy/core/expr.py]: Ok. I will change this.

**Comment by leosartaj** [comment]
I have renamed `differenceDelta`` to ``difference_delta``. Also, removed it from ``Expr``. @jcrist

**Comment by jcrist** [comment]
Looks good to me. Feel free to merge when tests pass.

**Comment by leosartaj** [comment]
Tests pass. Merging.

**Comment by leosartaj** [comment]
Thanks @jcrist

**Pull request merged**